src/main/java/org/candlepin/insights/task/TaskType.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.task;
+
+/**
+ * An enumeration representing the types of tasks that can be handled by rhsm-conduit.
+ */
+public enum TaskType {
```

Comment:
Did you consider just using `Class<? extends Task>` instead of defining an enum for this?

Reply:
Since the TaskDescriptor is the POJO describing a Task, I didn't want to reference classes as this could end up getting (de)serialized. It would prevent us from moving/renaming things later.